libraries/botframework-connector/src/auth/managedIdentityAuthenticator.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * @module botframework-connector
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AccessToken, DefaultAzureCredential } from '@azure/identity';
+import { retry } from 'botbuilder-stdlib';
+import { JwtTokenProviderFactoryInterface } from './jwtTokenProviderFactoryInterface';
+import { ok } from 'assert';
+
+/**
+ * Abstraction to acquire tokens from a Managed Service Identity.
+ */
+export class ManagedIdentityAuthenticator {
+    private readonly tokenProvider: DefaultAzureCredential;
+    private readonly resource: string;
+
+    /**
+     * Initializes a new instance of the ManagedIdentityAuthenticator class.
+     *
+     * @param appId Client id for the managed identity to be used for acquiring tokens.
+     * @param resource Resource for which to acquire the token.

```

Comment:
@msomanathan is there a more descriptive docstring we can use here? Or include `<remarks/>` that indicate what a `resource` might be?

Reply:
While not ideal, this comment is ported from .NET so it must be acceptable 😄